scripts/superdesk-authoring/metadata/metadata.js

```diff
@@ -469,6 +471,7 @@ function MetadataListEditingDirective(metadata, $filter) {
                             scope.activeTree = scope.tree[null];
                         } else {
                             scope.terms = _.clone(scope.activeTree) || [];
+                            scope.allSelected = (scope.item[scope.field].length === scope.list.length) ? true : false;
```

Comment:
`scope.allSelected = scope.item[scope.field].length === scope.list.length;`  should be enough